src/vsphere_cpi/lib/cloud/vsphere/vm_type.rb

```diff
@@ -0,0 +1,28 @@
+require 'cloud/vsphere/storage_list'
+
+module VSphereCloud
+  class VmType
+    include VSphereCloud::StorageList
+    attr_reader :datacenter, :storage_list
+
+    # @param [Resources::Datacenter] datacenter
+    # @param [Hash] cloud_properties
+    def initialize(datacenter, cloud_properties)
+      @datacenter = datacenter
+      @cloud_properties = cloud_properties
+      @storage_list = cloud_properties['datastores'] || []
```

Comment:
`@storage_list` should be a method because it'll go out of sync with `@cloud_properties` if `@cloud_properties` is mutable